app/src/main/java/com/example/timerppo/Activity/CreateWorkoutActivity.java

```diff
@@ -0,0 +1,259 @@
+            }
+        });
+        buttonRestPlus.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                createViewModel.setRestTime(createViewModel.getRestTime().getValue() + 1);
+            }
+        });
+        buttonWorkPlus.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                createViewModel.setWorkTime(createViewModel.getWorkTime().getValue() + 1);
+            }
+        });
+        buttonWorkMinus.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                createViewModel.setWorkTime(createViewModel.getWorkTime().getValue() - 1);
+            }
+        });
+        buttonSetPlus.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                createViewModel.setSets(createViewModel.getSets().getValue() + 1);
+            }
+        });
+        buttonSetMinus.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                createViewModel.setSets(createViewModel.getSets().getValue() - 1);
+            }
+        });
+
+        inputName.setOnKeyListener(new View.OnKeyListener() {
+            public boolean onKey(View v, int keyCode, KeyEvent event) {
+                if (event.getAction() == KeyEvent.ACTION_DOWN) {
+                    createViewModel.setName(inputName.getText().toString());
+                    return true;
+                }
+                return false;
+            }
+        });
+
+
+        findViewById(R.id.save).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Intent intent = getIntent();
+                Bundle bundle = intent.getExtras();
+                int[] id = (int[])bundle.get("timerId");
+
+                if (id[1] != 1) {
+                    IntegerHSLColor color = colorBar.getPickedColor();
+                    WorkoutModel workoutModel = new WorkoutModel(
+                            inputName.getText().toString(),
+                            Integer.parseInt(inputPrepair.getText().toString()),
+                            Integer.parseInt(inputWork.getText().toString()),
+                            Integer.parseInt(inputRest.getText().toString()),
+                            Integer.parseInt(inputCycle.getText().toString()),
+                            Integer.parseInt(inputSet.getText().toString()),
+                            Integer.parseInt(inputCalm.getText().toString()),
+                            Color.HSVToColor(new float[]{color.getFloatH(), color.getFloatL(), color.getFloatS()})
+                    );
+                    db.timerDao().insert(workoutModel);
```

Comment:
All this code goes to ViewModel. You shouldn't interact with DB from your view layer.